papis/docmatcher.py

```diff
@@ -186,8 +214,57 @@ def get_regex_from_search(search: str) -> Pattern[str]:
         properly escaped and allows for multiple spaces.
     """
     return re.compile(
-        ".*{}.*".format(".*".join(map(re.escape, search.split()))),
-        re.IGNORECASE)
+        ".*{}.*".format(".*".join(map(re.escape, search.split()))), re.IGNORECASE
+    )
+
+
+def check_syntax(parsed: List[ParseResult]) -> bool:
+    """Tests the syntax by replacing all search terms with True,
+    then trying to evaluate the resulting string.
+    """
+
+    test = []
+    for p in parsed:
+        if not p.syntax:
+            test.append("True")
+        else:
+            test.append(p.string)
+
+    try:
+        eval(" ".join(test))
+        return True
+    except SyntaxError:
+        logger.error("Malformed query.")
+        return False
+
+
+def proofread(parsed: List[ParseResult]) -> List[ParseResult]:
+    """Fixes a parsed query by inserting missing *and* operators.
+    For instance:
+    ' ein 192     photon' -> 'ein and 192 and photon'
+
+
+    :param a parsed a search string.
+    :returns: a fixed list of ParseResults or operators.
+    """
+    result = []
+    last = len(parsed) - 1
+    fixes = 0
+    for idx, token in enumerate(parsed):
+        result.append(token)
+        if idx != last and token.needsboolafter() and parsed[idx + 1].needsboolbefore():
+            # add 'and' when queries or syntax have have no operator in between
+            result.append(
+                ParseResult(
+                    syntax=True, string="and", pattern=get_regex_from_search("")
+                )
+            )
+            fixes += 1
+
+    if fixes > 0:
+        logger.debug("Fixed query by adding %s missing 'and' operator(s).", fixes)
```

Comment:
```suggestion
    logger.debug("Fixed query by adding %s missing 'and' operator(s).", fixes)
```
? Since it's a debug message, might as well always print it.

Reply:
Agree!